src/rgw/driver/sfs/sqlite/dbconn.h

```diff
@@ -256,11 +256,15 @@ inline auto _make_storage(const std::string& path) {
 using Storage = decltype(_make_storage(""));
 using StorageRef = Storage*;
 
+// TODO revisit this when code is fully ported to sqlite modern cpp
+using ConnectionNewLib = std::shared_ptr<sqlite3>;
+
 // TODO(https://github.com/aquarist-labs/s3gw/issues/788): Make
 // dbapi::sqlite::database the primary interface for sqlite3.
 class DBConn {
  private:
   std::unordered_map<std::thread::id, Storage> storage_pool;
+  std::unordered_map<std::thread::id, ConnectionNewLib> storage_pool_new;
```

Comment:
Just a general note:

It's a bit suboptimal that sqlite_modern_cpp want's a shared ptr. I'd rather have DBConn own all the sqlite connections with a unique_ptr. It's lifetime is already equal to sfs / RGW.

The nice thing is that with sqlite_modern_cpp's std::shared_ptr<sqlite3> constructor we can take full charge of the sqlite database initialization.